### PR TITLE
C++ Support

### LINKED ISSUE #17: C++ support

### PR DESCRIPTION
**Pull request by misson20000**
Closes #17. Credit to @dvdfreitag

### CODE REVIEW

File: link.T [hunk 1]
```diff
@@ -63,20 +63,18 @@ NORELOC_BSS_END_ = .;
...
     *(.eh_frame)
     __eh_frame_end = .;
   }
 
+  .eh_frame_hdr : ALIGN(0x1000) {
+    __eh_frame_hdr_start = .;
+    *(.eh_frame_hdr)
+    __eh_frame_hdr_end = .;
+  }
+
```
Comment (misson20000) [inline link.T]: These sections still don't make it into the final NRO. They get discarded by elf2nxo.py

Comment (dvdfreitag) [reply link.T]: Interesting, so the runtime errors we were seeing previously were simply an alignment issue, probably throwing _everything_ out of wack.

I assume this is some issue with elf2nxo, or nxodance?

**Comment by dvdfreitag** [comment]
Something in the merge with development broke the libcxx build, specifically it looks like an include path for `strtold` got broken.

**Comment by roblabla** [comment]
Related issue: https://github.com/reswitched/libtransistor/pull/98

**Comment by roblabla** [comment]
Investigation done by dvdfreitag:

```
dvdfreitag: So I think I know why exceptions aren't working
dvdfreitag: 

int square(int num)
{
    try
    {
        throw num;
    }
    catch (int a)
    {
        return a;
    }
}

dvdfreitag: That should produce the following output:
dvdfreitag: 

_Z6squarei:                             // @_Z6squarei
        str     x19, [sp, #-32]!        // 8-byte Folded Spill
        stp     x29, x30, [sp, #16]     // 8-byte Folded Spill
        add     x29, sp, #16            // =16
        mov      w19, w0
        orr     w0, wzr, #0x4
        bl      __cxa_allocate_exception
        str       w19, [x0]
        adrp    x1, :got:_ZTIi
        mov      x2, xzr
        ldr     x1, [x1, :got_lo12:_ZTIi]
        bl      __cxa_throw
        bl      __cxa_begin_catch
        ldr       w19, [x0]
        bl      __cxa_end_catch
        ldp     x29, x30, [sp, #16]     // 8-byte Folded Reload
        mov      w0, w19
        ldr     x19, [sp], #32          // 8-byte Folded Reload
        ret

dvdfreitag: But, in libtransistor all of __cxa_allocate_exception etc are replaced with #0
dvdfreitag: 

_Z6squarei:
    str    x19, [sp, #-32]!
    stp    x29, x30, [sp, #16]
    add    x29, sp, #16
    mov     w19, w0
    orr    w0, wzr, #0x4
    bl    #0
    str        w19, [x0]
    adrp    x1, #0
    mov     x2, xzr
    ldr        x1, [x1]
    bl    #0
    bl    #0
    ldr        w19, [x0]
    bl    #0
    ldp    x29, x30, [sp, #16]
    mov     w0, w19
    ldr    x19, [sp], #32
    ret

(edited)
dvdfreitag: Looks like libcxxabi.a doesn't contain any of those symbols :thinking:
dvdfreitag: Ah wait, I was wrong. They're there
dvdfreitag: mmm, interesting. When you link to an nro they are changed from zero
misson20000: (probably our link script being lame)
dvdfreitag: So __cxa_allocate_exception is replaced with #346276
dvdfreitag: Which is way out beyond the end of the nro, 5ba54 is the last address in text
dvdfreitag: Oh, actually
dvdfreitag: :face_palm:
dvdfreitag: So that's jumping to 548a4, which is somewhere in the middle of snprintf
dvdfreitag: __cxa_allocate_exception is in the nro, but at a totally different location
dvdfreitag: 

__cxa_allocate_exception:
   54960:

dvdfreitag: An offset of -188
dvdfreitag: Yeah it's not a fixed offset, __cxa_throw is -208 away from where it's actually at
dvdfreitag: And the type information symbols aren't present, eg _ZTIi which is out at 0x75000
dvdfreitag: Could this possibly be due to an issue with the modifications made to nxodance?
misson20000: It sounds like an nxodance issue, yeah
dvdfreitag: :/
dvdfreitag: 

Function                    Address        Real Address    Offset
__cxa_allocate_exception    0x548A4        0x54960         -0xBC
__cxa_throw                 0x5493C        0x54A0C         -0xD0
__cxa_begin_catch           0x54A24        0x54AF8         -0xD4
__cxa_end_catch             0x54AB8        0x54b94         -0xDC


dvdfreitag: Retr0id roblabla Looks like the issue with exceptions is something to do with the modifications that were made to nxodance ^
```

**Pull request merged**